source/usr/lib/mcvirt/cluster/cluster.py

```diff
@@ -15,437 +17,555 @@
 
             # Set the VM node
-            remote_object.runRemoteCommand('virtual_machine-setNode',
-                                           {'vm_name': vm_object.getName(),
-                                            'node': vm_object.getNode()})
+            remote_virtual_machine_object.setNodeRemote(vm_object.getNode())
+
+    def check_remote_machine(self, remote_connection):
+        """Perform checks on the remote node to ensure that there will be
+        no object conflicts when syncing the Network and VM configurations
+        """
+        # Ensure that the remote node has no cluster nodes
+        remote_cluster = remote_connection.get_connection('cluster')
+        if len(remote_cluster.get_nodes(return_all=True)):
+            raise RemoteObjectConflict('Remote node already has nodes attached')
 
-    def checkRemoteMachine(self, remote_object):
-        """Performs checks on the remote node to ensure that there will be
-           no object conflicts when syncing the Network and VM configurations"""
         # Determine if any of the local networks/VMs exist on the remote node
-        remote_networks = remote_object.runRemoteCommand('node-network-getConfig', [])
-        from mcvirt.node.network import Network
-        for local_network in Network.getConfig().keys():
-            if (local_network in remote_networks.keys()):
-                raise RemoteObjectConflict('Remote node contains duplicate network: %s' %
-                                           local_network)
-
-        from mcvirt.virtual_machine.virtual_machine import VirtualMachine
-        local_vms = VirtualMachine.getAllVms(self.mcvirt_instance)
-        remote_vms = remote_object.runRemoteCommand('virtual_machine-getAllVms', [])
-        for local_vm in local_vms:
-            if (local_vm in remote_vms):
-                raise RemoteObjectConflict('Remote node contains duplicate Virtual Machine: %s' %
-                                           local_vm)
-
-        # If DRBD is enabled on the local machine, ensure it is installed on the remote machine
+        remote_network_factory = remote_connection.get_connection('network_factory')
+
+        # Check that each of the interfaces, used for the networks, is present on the
+        # remote node
+        network_factory = self._get_registered_object('network_factory')
+        for local_network in network_factory.get_all_network_objects():
+            if not remote_network_factory.interface_exists(local_network.get_adapter()):
+                raise RemoteObjectConflict('Network interface %s does not exist on remote node' %
+                                           local_network.get_adapter())
+
+        # Determine if there are any VMs on the remote node
+        remote_virtual_machine_factory = remote_connection.get_connection(
+            'virtual_machine_factory')
+        if len(remote_virtual_machine_factory.getAllVirtualMachines()):
```

Comment:
in python empty arrays evaluate to false, in my opinion it is both shorter to write and easier to read if this line is replaced by

`if remote_virtual_machine_factory.getAllVirtualMachines():`